enzyme/Enzyme/TypeAnalysis/TypeTree.h

```diff
@@ -731,6 +735,8 @@ class TypeTree : public std::enable_shared_from_this<TypeTree> {
         chunk = 8;
       } else if (flt->isHalfTy()) {
         chunk = 2;
+      } else if (flt->isX86_FP80Ty()) {
```

Comment:
Same comment here about `chunk = dl.getTypeSizeInBits(flt) / 8;`

Reply:
I couldn't simplify the code here like at the other places because there is no `DataLayout` instance (as far as I can see).

Reply:
What would it look like to add an argument for a datalayout here?

Reply:
In every place where `IsAllFloat` is called, there is a DataLayout instance lying around. In 6a707777e364fd060d6f21da78f0ed371322c7f5, I add a DataLayout argument to the signature of IsAllFloat, in order to simplify its implementation in the same way as in the other places. (Could you please take another look and confirm that this change is correct?)